send2cgeo.user.js

```diff
@@ -25,107 +24,130 @@
+    // hide iframe for now and wait for page to be loaded
+    + '$("#send2cgeo iframe")'
+    + '   .hide()'
+    + '   .off("load")'
+    + '   .attr("src", "https://send2.cgeo.org/add.html?cache=" + GCCode)'
+    + '   .on("load",'
+    + '       function() {'
+                  // hide "please wait text" and show iframe
+    + '           $("#send2cgeo div").hide();'
+                  // hide box after 3 seconds
+    + '           $(this).css("display", "block").parent().delay(3000).fadeOut();'
+    + '       }'
+    + '   );'
+    + '};';
+
+document.getElementsByTagName('head')[0].appendChild(s2cgScript);
+
+var s2geomultiScript = document.createElement('script');
+s2geomultiScript.type = 'text/javascript';
+s2geomultiScript.innerHTML = ''
+    + 'window.s2geomulti = function(requestedCnt, skipFound, moreClicked) {'
+    + '    if (typeof(skipFound) == "undefined" ) {'
+    + '        skipFound = $("#send2cgeo_skip_found").is(":checked");'
+    + '    }'
+    + '    var alreadySent = $("[send2cgeo_sent]").length;'
+    + '    if (alreadySent < requestedCnt) {'
+    + '        var toAddIframe = $("[send2cgeo_gccode]").not("[send2cgeo_sent]").first();'
+    + '        if (toAddIframe.length) {'
+    + '            s2geomultiProcessLine(toAddIframe, skipFound, requestedCnt, alreadySent);'
+    + '        } else {'
+    + '            s2geomultiLoadMore(requestedCnt, alreadySent, skipFound, moreClicked);'
+    + '        }'
+    + '    } else {'
+    + '        s2cgeoProgressReport("Finished after sending " + alreadySent + " caches.");'
+    + '    }'
+    + '};'
+
+    + 'function s2cgeoProgressReport(message, append) {'
+    + '    if (append) {'
+    + '        $("#send2cgeo_controls_div").append(message);'
+    + '    } else {'
+    + '        $("#send2cgeo_controls_div").html(message);'
+    + '    }'
+    + '}'
+
+    + 'function s2geomultiLoadMore(requestedCnt, alreadySent, skipFound, moreClicked) {'
+    + '    if (!moreClicked) {'
+    + '        s2cgeoProgressReport("Waiting to load more caches from web site, sent " + alreadySent + " caches.", false);'
+    + '        var loadMoreBtn = $("#loadmore");'
+    + '        loadMoreBtn.click();'
+    + '        moreClicked = true;'
+    + '    }'
+    + '    s2cgeoProgressReport(".", true);'
+    + '    setTimeout('
+    + '        function() {'
+    + '            s2geomulti(requestedCnt, skipFound, moreClicked);'
+    + '        },'
+    + '        2000'
+    + '    );'
+    + '}'
+
+    + 'function s2geomultiProcessLine(toAddIframe, skipFound, requestedCnt, alreadySent) {'
+    + '    if (skipFound) {'
+    + '        var GCCode = toAddIframe.attr("send2cgeo_gccode");'
+    + '        toAddIframe.html("<iframe width=120 height=80 src=\'https://send2.cgeo.org/add.html?cache=" + GCCode + "\'>");'
+    + '        toAddIframe.attr("send2cgeo_sent", "1");'
+    + '        s2cgeoProgressReport((alreadySent + 1) + " ", alreadySent !== 0);'
+    + '    }'
```

Comment:
original:
~~~
  function s2geomultiProcessLine(toAddIframe, skipFound, requestedCnt, alreadySent) {
      if (skipFound) {
          if (toAddIframe.parent().parent().find("use[xlink\\:href*=#icon-found]").length === 0) {
              var code = toAddIframe.attr("send2cgeo_gccode");
              toAddIframe.html("<iframe width=120 height=80 src=\"https://send2.cgeo.org/add.html?cache="+code+"\">");
              toAddIframe.attr("send2cgeo_sent","1");
              s2cgeoProgressReport((alreadySent+1)+" ",alreadySent!==0);
          } else {
              var trToDel = toAddIframe.parent().parent();
              trToDel.remove();
              s2cgeoProgressReport(".",alreadySent!==0);
          }
      }
~~~
Which problem was solved by skipping some lines here?

Reply:
Maybe it was lost during copy and paste ... Sorry